src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetService.cs

```diff
@@ -122,23 +126,65 @@ internal class DataSetService(
 
         var totalResults = await queryable.CountAsync(cancellationToken: cancellationToken);
 
-        var dataSetVersions = (await queryable
-                .OrderByDescending(dsv => dsv.VersionMajor)
-                .ThenByDescending(dsv => dsv.VersionMinor)
-                .Skip((page - 1) * pageSize)
-                .Take(pageSize)
-                .ToListAsync(cancellationToken: cancellationToken)
-            )
-            .Select(MapDataSetVersion)
+        var dataSetVersions = await queryable
+            .OrderByDescending(dsv => dsv.VersionMajor)
+            .ThenByDescending(dsv => dsv.VersionMinor)
+            .Skip((page - 1) * pageSize)
+            .Take(pageSize)
+            .ToListAsync(cancellationToken: cancellationToken);
```

Comment:
I think you can use `.Paginate(page: page, pageSize: pageSize)` here if you want.